pkg/kubectl/rolling_updater.go

```diff
@@ -353,49 +404,101 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
 		}
 	}
 
-	// +1, -1 on oldRc, newRc until newRc has desired number of replicas or oldRc has 0 replicas
-	for newRc.Spec.Replicas < desired && oldRc.Spec.Replicas != 0 {
-		newRc.Spec.Replicas += 1
-		oldRc.Spec.Replicas -= 1
-		fmt.Printf("At beginning of loop: %s replicas: %d, %s replicas: %d\n",
-			oldName, oldRc.Spec.Replicas,
-			newName, newRc.Spec.Replicas)
-		fmt.Fprintf(out, "Updating %s replicas: %d, %s replicas: %d\n",
-			oldName, oldRc.Spec.Replicas,
-			newName, newRc.Spec.Replicas)
+	oldSpec := &oldRc.Spec
+	newSpec := &newRc.Spec
 
-		newRc, err = r.scaleAndWait(newRc, retry, waitForReplicas)
-		if err != nil {
-			return err
+	// Compute the scale amount based on a percentage of the new desired count.
+	// A negative percentage indicates our scale direction should be down-first.
+	scaleAmount := 1
+	scaleUpFirst := true
+	if config.UpdatePercent != nil {
+		scaleAmount = int(math.Ceil(float64(desired) * (math.Abs(float64(*config.UpdatePercent)) / 100)))
+		if *config.UpdatePercent < 0 {
+			scaleUpFirst = false
 		}
-		time.Sleep(updatePeriod)
-		oldRc, err = r.scaleAndWait(oldRc, retry, waitForReplicas)
-		if err != nil {
-			return err
+	}
+	// Helpful output about what we're about to do.
+	direction := "up"
+	if !scaleUpFirst {
+		direction = "down"
+	}
+	fmt.Fprintf(out, "Scaling up %s from %d to %d, scaling down %s from %d to 0 (scale %s first by %d each interval)\n",
+		newName, newSpec.Replicas, desired, oldName, oldSpec.Replicas, direction, scaleAmount)
+
+	// Scale newRc and oldRc until newRc has the desired number of replicas and
+	// oldRc has 0 replicas.
+	for newSpec.Replicas != desired || oldSpec.Replicas != 0 {
+		// Set up a function for scaling up.
+		scaleUp := func() error {
+			// Nothing to do.
+			if newSpec.Replicas == desired {
+				return nil
+			}
+			// Scale up new. Account for fencepost conditions. If the old spec is
+			// already scaled to 0, go ahead and scale all the way up.
+			newSpec.Replicas += scaleAmount
+			if newSpec.Replicas > desired || oldSpec.Replicas == 0 {
+				newSpec.Replicas = desired
+			}
+			fmt.Fprintf(out, "Scaling %s up to %d\n", newName, newSpec.Replicas)
+			newRc, err = r.scaleAndWait(newRc, retry, waitForReplicas)
+			if err != nil {
+				return err
+			}
+			newSpec = &newRc.Spec
+			// Perform acceptance checking, if necessary.
+			if config.UpdateAcceptor != nil {
```

Comment:
You should have a no-op acceptor, and require the acceptor to print its own message.  Also, I'm not sure you really need to print out "Performing acceptance check".  That would reduce this to `if err := config.UpdateAcceptor.Accept(newRc); err != nil { return ... }`